packages/firestore/src/local/local_store.ts

```diff
@@ -709,52 +709,50 @@ export class LocalStore {
   }
 
   /**
-   * Unpin all the documents associated with the given query and remove its
-   * query data from the query cache.
+   * Unpin all the documents associated with the given query. If
+   * `keepPersistedQueryData` is set to false and Eager GC enabled, the method
+   * directly removes the associated query data from the query cache.
    */
-  releaseQuery(query: Query): Promise<void> {
-    return this.persistence.runTransaction('Release query', true, txn => {
-      return this.queryCache
-        .getQueryData(txn, query)
-        .next((queryData: QueryData | null) => {
-          assert(
-            queryData != null,
-            'Tried to release nonexistent query: ' + query
-          );
-          this.localViewReferences.removeReferencesForId(queryData!.targetId);
-          delete this.targetIds[queryData!.targetId];
-          if (this.garbageCollector.isEager) {
-            return this.queryCache.removeQueryData(txn, queryData!);
-          } else {
-            return PersistencePromise.resolve();
-          }
-        })
-        .next(() => {
-          // If this was the last watch target, then we won't get any more
-          // watch snapshots, so we should release any held batch results.
-          if (objUtils.isEmpty(this.targetIds)) {
-            const documentBuffer = new RemoteDocumentChangeBuffer(
-              this.remoteDocuments
-            );
-            return this.releaseHeldBatchResults(txn, documentBuffer).next(
-              () => {
-                documentBuffer.apply(txn);
-              }
+  // PORTING NOTE: `keepPersistedQueryData` is multi-tab only.
+  releaseQuery(query: Query, keepPersistedQueryData: boolean): Promise<void> {
+    const requirePrimaryLease = keepPersistedQueryData === false;
```

Comment:
FWIW I verified that if you change this to 
```
const requirePrimaryLease = this.garbageCollector.isEager;
```

and you remove the `!keepPersistedQueryData` check below, the tests all still pass.  So I still think we don't actually _need_ the `keepPersistedQueryData` argument.  But you can leave it if you think it's more correct or we'll need it in the future.

Reply:
In the multi-tab spec tests, this method only ever seems to be called this `this.garbageCollector.isEager` set to `false`. I don't necessarily want to rely on this, or use this state to toggle unrelated behavior. I prefer to be explicit.